### DISCUSSION TITLE
Floating UI: set reference

### DISCUSSION THREAD

**Discussion by Lighttree** [floating-ui/floating-ui#1439] [Q&A] [answered]
Hi guys !

I'm working on the internal `popper` component. For proprietary design system.
I have 2 questions.

1. Is it future proof to use `react-popper` for this purpose? Or it's better to start using `useFloating` from `@floating-ui/react-dom` ?

2. At the moment I'm going to build the component based on `useFloating`, but I want to introduce API that allows consumers to pass referer like this:

```jsx
<Popover anchor={ref}>Content</Popover>
```

> For better back-compatibility

As far as I can see `useFloating` returns `reference` by itself. But I have issues with setting it to another ref that was passed into my `<Popover />` it seems `reference` is read-only.

Is it possible to change this `reference` to `ref` that was passed into the component?
Or I must change my API to something like

```jsx
const {reference, Popover} = usePopover()
```
and ask the consumer to put returned `reference` to an element that should be an "anchor"?

**Comment by atomiks** [comment, marked as answer]
1. For new components/apps, `useFloating` is the way forward, `react-popper` is low maintenance mode now
2. `reference` is just a callback ref function. It's possible to compose many different refs, but I'm not sure I fully understand your question?

```js
<div ref={node => reference(node)} />
```

**Comment by Lighttree** [comment]
@atomiks Thanks. Sorry I missed the point that reference is callback ref function, not a `ref` itself.

I was able to do what I wanted by doing:
```js
// anchor.current - an element that I'm passing into my component.
useEffect(() => {
    reference(anchor.current)
 }, [])
 ```
